### PR TITLE
Implement kronecker_symbol

### LINKED ISSUE #998: Legendre, Jacobi and Kronecker symbols
First of all, we don't have the Legendre and Kronecker symbols, because flint does not have it I guess.

Now the `fmpz_jacobi` docstring says that for `(a, b)` the number  `b` must be a prime. But this would make it the Legendre symbol? On the other hand, `fmpz_jacobi` is calling `mpz_jacobi`, which really is the Jacobi symbol (see https://gmplib.org/manual/Number-Theoretic-Functions) and thus is restricted to `b` odd, but not necessarily prime. There is some discrepancy between the docstring and the implementation. Not sure if we can just change the docstring. For this one would have to check what `n_jacobi` is doing.

What should we do about this? Should I move this issue to the flint repository?

### PR DESCRIPTION
**Pull request by wbhart**
Fixes #998

### CODE REVIEW

**Review by tthsqe12 — COMMENTED** [review]
First of all, the kronecker symbol has no restriction on the input. Second, is there a reason to avoid fmpz_kronecker?

**Comment by wbhart** [comment]
What is the definition when the modulus is 0?

**Comment by wbhart** [comment]
Never heard that definition before.

**Comment by tthsqe12** [comment]
Sorry, i fixed it

**Comment by wbhart** [comment]
That's not what the function returns. Do you have a reference for your definition?

**Comment by tthsqe12** [comment]
The test code tests against mpz_kronecker as a reference value, and the mpz docs reference cohen.

**Comment by thofma** [comment]
It is Cohen, the first book, Definition 1.4.8.

**Comment by wbhart** [comment]
Thanks, I added the reference, as there are lots of different definitions out there.

File: docs/src/integer.md [hunk 1]
```diff
@@ -362,6 +362,7 @@ moebius_mu(::fmpz)
 ```@docs
 jacobi_symbol(::Int, ::Int)
 jacobi_symbol(::fmpz, ::fmpz)
+kronecker_symbol(::fmpz, ::fmpz)
```
Comment (thofma) [inline docs/src/integer.md:365]: There is no docstring for this. We could attach the docstring to
```
function kronecker_symbol end
```
Then this should work.

Comment (wbhart) [reply docs/src/integer.md:365]: Oh, I'll just correct the place where it is instantiated.

**Comment by tthsqe12** [comment]
Ok, if you are happy with this so far, then kronecker_symbol(::Int,::Int) is implemented in z_kronecker, which also uses mpz_kronecker as a test reference.

**Review by tthsqe12 — APPROVED** [review]

**Pull request merged**